compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -2055,6 +2055,28 @@ void OMR::X86::CodeGenerator::doBinaryEncoding()
             }
          }
 
+      if (estimateCursor->getNext() == NULL && estimateCursor->getKind() != TR::Instruction::IsPadding)
+         {
+         TR::Instruction *lastInstruction = estimateCursor;
+
+         while (lastInstruction != NULL)
+            {
+            if (!lastInstruction->getOpCode().isPseudoOp())
+               break;
+
+            lastInstruction = lastInstruction->getPrev();
+            }
+
+         // In some cases, the last instruction in the function will be a call instruction. This may happen
+         // when throwing an exception. The return address of the function may be beyond the body of this
+         // function. In this case, we must add padding to end of the function to prevent the stack walker
+         // from detecting an illegal return address.
```

Comment:
This comment suggests this padding is wholly a J9-specific requirement that does not apply to other consumers of OMR.  Taken together with my earlier comment, perhaps the way this should be implemented is to a "postBinaryEstimationWalk()" function (or some other creatively named thing) that is called after the estimation loop.  Downstream projects can implement what they need in it.  The OMR implementation can do nothing for now.

Reply:
I'll close this and move this logic to OpenJ9. We can do it in the function prologue generation.